### ISSUE TITLE
Hash values different for same Country obj

### ISSUE THREAD

**Issue by anithri** [hexorx/countries#400]
```
c1 = ISO3166::Country.new("US")
c2 = ISO3166::Country.new("US")
c1 == c2 #true
c1.hash == c2.hash #false
hsh = {}
hsh[c1] = 1
hsh[c2] = 2
hsh.keys.count # 2
```

This doesn't seem like the correct behavior.  It makes useful things like `group_by(&:country)` useless.

**Comment by rposborne** [comment]
@anithri please review #401
Does this cover your use case?

**Comment by rposborne** [comment]
Closed by #401
